qcodes/instrument_drivers/oxford/fridge_http_server.py

```diff
@@ -0,0 +1,231 @@
+
+                return web.Response(text=str(data))
+            elif attribute == None:
+                return web.Response(status=404, text="Usage ip/parameter?attribute=attributes i.e. "
+                                                     "ip/T1/attribute=value. "
+                                                     "Valid attributes are {}".format(valid_attributes))
+            else:
+                return web.Response(status=404, text="Parameter {} does not have attribute {}".format(parameter, attribute))
+        else:
+            return web.Response(status=404, text="Parameter {} not found".format(parametername))
+
+    async def index(self, request):
+        """
+        Handler for a very basic index page that just tells you how to use the html api.
+        """
+        return web.Response(text="Usage ip/parameter?attribute=value i.e. ip/T1/attribute=value")
+
+    async def handle_hostname(self, request):
+        host =  socket.gethostname()
+        return web.Response(text=host)
+
+    @staticmethod
+    def prepare_monitor_data(triton) -> Dict[str, Union[list, float]]:
+        """
+        Return a dict that contains the parameter from the Triton to be monitored.
+        """
+
+        def get_data(triton, parametername):
+            parameter = getattr(triton, parametername)
+            parameter.get()
+            temp = parameter._latest()
+            temp['unit'] = parameter.unit
+            # replace with reading directly from reg database once running on fridge computer
+            namedict = triton.chan_temp_names.get(parametername, None)
+            if namedict:
+                fullname = namedict['name']
+            else:
+                fullname = None
+            temp['name'] = fullname or parameter.label or parameter.name
+            temp['value'] = str(temp['value'])
+            if temp["ts"] is not None:
+                temp["ts"] = time.mktime(temp["ts"].timetuple())
+
+            return temp
+
+        triton_parameters = []
+        for parametername in triton.parameters:
+            temp = get_data(triton, parametername)
+            triton_parameters.append(temp)
+
+        triton_parameters.append(get_data(triton, 'action'))
+        triton_parameters.append(get_data(triton, 'status'))
+
+        parameters_dict = {"instrument": triton.name, "parameters": triton_parameters}
+        state = {"ts": time.time(), "parameters": [parameters_dict]}
+
+        return state
+
+    @staticmethod
+    def prepare_ws_data(triton) -> Dict[str, Union[list, float]]:
+        """
+        Return a dict that contains the parameter from the Triton to be exported via WS. This includes a read only copy
+        of all parametes exposed by the driver.
+        """
+
+        triton_parameters = []
+        for parametername in triton.parameters:
+            parameter = getattr(triton, parametername)
+            temp = {}
+            temp['value'] = parameter.get()
```

Comment:
Maybe, not sure. If it's a get_latest and nothing else is reading from the remote driver then it will never be updated?